api/python/provisioner/commands/remove_swupgrade_repo.py

```diff
@@ -0,0 +1,54 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import logging
+from typing import Type
+
+from provisioner import ALL_MINIONS
+
+from .set_swupgrade_repo import SetSWUpgradeRepo
+from .. import inputs
+
+logger = logging.getLogger(__name__)
+
+
+class RemoveSWUpgradeRepo(SetSWUpgradeRepo):
+
+    input_type: Type[inputs.SWUpgradeRemoveRepo] = inputs.SWUpgradeRemoveRepo
+
+    def run(self, *args, targets=ALL_MINIONS, dry_run=False, **kwargs):
+        # static validation
+        if len(args) == 1 and isinstance(args[0], self.input_type):
```

Comment:
Can we make that new API as an only way how to cleanup a repo?

Currently similar logic might be triggeret in scope of set command using UNDEFINED. I believe makes sense:

- drop such possibility (or at least deprecate that with some WARNING message).
- for candidate repo [cleanup](https://github.com/Seagate/cortx-prvsnr/blob/29e5a0e5d5b121e59d7d8c382f21b2863b2a208a/api/python/provisioner/commands/set_swupgrade_repo.py#L329-L332) call that API instead

Reply:
I need clarification here:
Do you want to change behavior here, in the `RemoveSWUpgradeRepo.run` method or you want to change the parent's (`Set`) class behavior? If the last one, this change will affect the `set_swupdate_repo` logic and we decided to not touch the old R1 API behavior.

Reply:
Yes it will affect old logic but only its internal part but API won't be impacted. Please create a Jira follow-up task for that and mention it here.

Reply:
I created the corresponding ticket in Jira
https://jts.seagate.com/browse/EOS-19243